opentelemetry-example/src/main/scala/zio/telemetry/opentelemetry/example/http/Client.scala

```diff
@@ -1,32 +1,29 @@
 package zio.telemetry.opentelemetry.example.http
 
-import sttp.client3._
-import sttp.client3.ziojson._
-import sttp.model.Uri
-import zio.telemetry.opentelemetry.example.config.AppConfig
 import zio._
+import zio.http.{ Header, Headers, Request, URL }
+import zio.json._
 import zio.telemetry.opentelemetry.example.Backend
+import zio.telemetry.opentelemetry.example.config.AppConfig
 
 case class Client(backend: Backend, config: AppConfig) {
 
   private val backendUrl =
-    Uri
-      .safeApply(config.backend.host, config.backend.port)
-      .map(_.withPath("status"))
+    URL
+      .decode(s"http://${config.backend.host}/${config.backend.port}")
       .left
       .map(new IllegalArgumentException(_))
 
   def status(headers: Map[String, String]): Task[Statuses] =
     for {
       url      <- ZIO.fromEither(backendUrl)
-      response <- backend
-                    .send(
-                      basicRequest
-                        .get(url.withPath("status"))
-                        .headers(headers)
-                        .response(asJson[Status])
-                    )
-      status    = response.body.getOrElse(Status.down("backend"))
+      request   = Request
+                    .get(url)
+                    .copy(headers = Headers(headers.map(x => Header.Custom(x._1, x._2))))
+      response <- backend.request(request)
+      status   <- JsonDecoder[Status]
+                    .decodeJsonStream(response.body.asStream.map(_.toChar))
+                    .catchAll(_ => ZIO.succeed(Status.down("backend")))
```

Comment:
How about rewriting it to the following:
```
      json     <- response.body.asString(StandardCharsets.UTF_8)
      status   <- ZIO.fromEither(JsonDecoder[Status].decodeJson(json)).catchAll(_ => ZIO.succeed(Status.down("backend")))
```

Reply:
fixed